RELEASE_NOTES.md

```diff
@@ -68,6 +68,13 @@ v1.0.0-alpha.xx
   objects, respectively.
   [#914](https://github.com/OpenAssetIO/OpenAssetIO/issues/914)
 
+### Bug fixes
+
+- Reintroduced the optional optimized entity reference prefix check
+  for `isEntityReferenceString`.
```

Comment:
[fix] Can we elaborate that it will short-circuit python?

Reply:
Elaborated in ab9b2c84, how's that?

Reply:
Epic, thanks!